WordPress/gutenberg
# Allow wrapping for Navigation block links

## Description
Fixes #18337.

Wraps text within a navigation link.

Links already had a `width: max-content;` rule, which prevented wrapping, so I had to remove this.

I also noticed when wrapping occurs, the submenu item indicator is positioned in an unusual way (it was always positioned on the right side of the block), so this PR also fixes that by changing the location of the indicator in the DOM. The indicator is now brought inline with the text.

There's an issue in IE11 where the indicator wraps to the next line (see screenshot). I haven't been able to work out how to fix that. Advice/commits are welcome. I don't think it's any worse than it was and it only occurs when the line wraps—here's a non-wrapping line:
<img width="520" alt="Screenshot 2020-04-16 at 11 34 48 am" src="https://user-images.githubusercontent.com/677833/79411792-512f3a80-7fd6-11ea-8937-4b68f6652475.png">


## How has this been tested?
1. Add a navigation block
2. Add a navigation link and type in some very long text (preferably a very long word)
3. See the wrapping

## Screenshots 
#### Before (Firefox)
<img width="591" alt="Screenshot 2020-04-16 at 11 04 49 am" src="https://user-images.githubusercontent.com/677833/79410133-29d66e80-7fd2-11ea-932d-f87fa2db177a.png">

#### After (Firefox)
<img width="558" alt="Screenshot 2020-04-16 at 10 59 16 am" src="https://user-images.githubusercontent.com/677833/79410195-4d99b480-7fd2-11ea-94b8-cfd16d3c6765.png">

#### Before (IE 11)
<img width="522" alt="Screenshot 2020-04-16 at 11 28 16 am" src="https://user-images.githubusercontent.com/677833/79411455-740d1f00-7fd5-11ea-8bd6-77dda255fbc7.png">

#### After (IE 11)
<img width="522" alt="Screenshot 2020-04-16 at 11 26 01 am" src="https://user-images.githubusercontent.com/677833/79411479-7ff8e100-7fd5-11ea-87e5-a9799c1461a0.png">



## Types of changes

Bug fix (non-breaking change which fixes an issue)



## Checklist:
